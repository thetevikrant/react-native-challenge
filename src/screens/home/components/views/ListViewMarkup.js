import * as React from "react";
import {
  Text,
  Body,
  List,
  ListItem,
  Thumbnail,
} from "native-base";

import styles from "../styles";
export interface Props {
  navigation: any;
  list: Object;
}
export interface State {}
class ListViewMarkup extends React.Component<Props, State> {
  render() {
    const { list, navigation } = this.props;
    return (
          <List>
            {list.map((item, index) => (
            <ListItem
              key={index}
              onPress={() =>
                navigation.navigate("Details", {
                  data: { item }
                })}
            >
              <Thumbnail square size={80} source={{ uri: item.img }} />
              <Body>
                <Text style={styles.title}>{item.title}</Text>
                <Text style={styles.description} note numberOfLines={2}>{item.description}</Text>
                <Text note>{item.publish_date}</Text>
              </Body>
            </ListItem>
            ))}
          </List>
    );
  }
}

export default ListViewMarkup;
