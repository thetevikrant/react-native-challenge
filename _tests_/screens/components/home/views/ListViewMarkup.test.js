import React from "react";
import ListViewMarkup from "../../../../../src/screens/home/components/views/ListViewMarkup";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

const navigation = { navigate: jest.fn() };
const list = { map: jest.fn() };

it("renders correctly", () => {
  const tree = renderer.create(<ListViewMarkup navigation={navigation} list={list} />).toJSON();
  expect(tree).toMatchSnapshot();
});