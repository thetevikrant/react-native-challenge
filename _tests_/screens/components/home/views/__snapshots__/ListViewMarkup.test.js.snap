// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  closeOnRowBeginSwipe={false}
  closeOnRowPress={true}
  closeOnScroll={true}
  directionalDistanceChangeThreshold={2}
  disableLeftSwipe={false}
  disableRightSwipe={false}
  leftOpenValue={0}
  previewFirstRow={false}
  recalculateHiddenLayout={false}
  rightOpenValue={0}
  style={Object {}}
  swipeToOpenPercent={50}
  virtual={undefined}
/>
`;
